### DISCUSSION TITLE
Firebase Messaging Dashboard shows 0 received, 0 opened, 0 impressions

### DISCUSSION THREAD

**Discussion by dJani97** [firebase/flutterfire#8028] [Q&A]
Hi!

We've been using FCM in our Flutter app for a while, having sent 6.5K notifications according to the Firebase Messaging Dashboard.

This seems reasonable. But the problem is, the dashboard shows 0 received, 0 opened and 0 impressions, which is weird, given that our team is constantly testing and improving push notifications, so they are definitely opening a large fraction of the messages.

In my code I am listening to `onMessage`, `onMessageOpenedApp`, and `getInitialMessage`.

Do I have to do something else in my code to report back to FCM that the client has received and/or read the message?

**Comment by alloy2rivet** [comment]
I've found the exact same problem,

REMOTE_NOTIFICATION_RECEIVED =0
REMOTE_NOTIFICATION_DISPLAY =0
REMOTE_NOTIFICATION_OPEN =0
